build/v1/types.go

```diff
@@ -902,11 +902,13 @@ type BuildConfigSpec struct {
 	CommonSpec `json:",inline" protobuf:"bytes,3,opt,name=commonSpec"`
 
 	// successfulBuildsHistoryLimit is the number of old successful builds to retain.
-	// If not specified, all successful builds are retained.
+	// By default the 5 most recent successful builds are retained.
+	// If removed, all successful builds are retained.
```

Comment:
this is slightly confusing.  

you have to remove it *after* you have gone through a create flow.

people might think they can remove this in yaml and do an oc create and get "All builds retained" behavior, which is not the case.

Reply:
fyi the magic is here: https://github.com/openshift/origin/blob/432c25701e4c5f3bc7e373715d25e53cdcc48956/pkg/build/apiserver/registry/buildconfig/strategy.go#L90-L103

Reply:
(so you can see how the default is injected only on create and only if the field is nil)